test/index.js

```diff
@@ -634,3 +634,53 @@ describe('views()', function () {
         done();
     });
 });
+
+describe('Plugin', function () {
+
+    it('can be registered before connections', function (done) {
+
+        var plugin = function (server, options, next) {
+
+            server.dependency('vision');
+            server.connection();
+            next();
+        };
+
+        plugin.attributes = {
+            connections: false,
+            name: 'test'
+        };
+
+        var server = new Hapi.Server();
+        server.register([Vision, plugin], Hoek.ignore);
+
+        expect(server.views).to.be.a.function();
+        server.initialize(function (err) {
+
+            if (err) {
+                return done(err);
+            }
+
+            server.stop(done);
+        });
+    });
+
+    it('only registers once', function (done) {
```

Comment:
It seemed unnecessary to me since when the plugin registration fails it passes an error to `done()` which in turn causes the test to fail. I'm fine adding an explicit `expect` though if this isn't explicit enough.